From 90940a1c91a998892b54b7742d6eb22e7262f640 Mon Sep 17 00:00:00 2001
From: Andreas Stefl <stefl.andreas@gmail.com>
Date: Sat, 3 May 2025 14:24:09 +0200
Subject: [PATCH 2/2] use env AR if set

---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index cb991bb..486048e 100644
--- a/Makefile
+++ b/Makefile
@@ -124,7 +124,7 @@ LINKED_LIB_SH := lib$(LIB_NAME).$(LINKED_LIB_EXT)
 endif
 
 # Some systems don't provide an unprefixed ar when cross-compiling.
-AR=ar
+AR ?= ar
 
 LIBRARIES = $(LIB_SH) $(LIB_ST)
 HEADERS = include/argon2.h
-- 
2.39.5 (Apple Git-154)

