diff --git a/CMakeLists.txt b/CMakeLists.txt
index 7c097cb..7553706 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -192,6 +192,15 @@ if (ODR_TEST)
     add_subdirectory(test)
 endif ()
 
+install(
+        DIRECTORY include/
+        DESTINATION include/
+        )
+install(
+        DIRECTORY src/
+        DESTINATION include/
+        FILES_MATCHING PATTERN "*.h"
+        )
 install(
         TARGETS
         odr