--- wvWare.c
+++ wvWare.c
@@ -1604,6 +1604,15 @@
     return (0);
 }
 
+const char *get_data_dir()
+{
+  const char *data_dir = getenv("WVDATADIR");
+  if (NULL == data_dir) {
+    data_dir = WVDATADIR;
+  }
+  return data_dir;
+}
+
 int
 wvOpenConfig (state_data *myhandle,char *config)
 {
@@ -1618,7 +1627,7 @@
 
     if(tmp == NULL)
     {
-	str_copy  (buf, sizeof(buf), WVDATADIR);
+	str_copy  (buf, sizeof(buf), get_data_dir());
 	str_append(buf, sizeof(buf), "/");
 	str_append(buf, sizeof(buf), config);
 	config = buf;
