diff --git a/CMakeLists.txt b/CMakeLists.txt
index ad6cedd..45d26a3 100755
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -16,6 +16,8 @@ if(${OPENLIBM_ARCH_FOLDER} STREQUAL "amd64" OR ${OPENLIBM_ARCH_FOLDER} STREQUAL
     set(OPENLIBM_ARCH_FOLDER "amd64")
 elseif(${OPENLIBM_ARCH_FOLDER} STREQUAL "arm64" OR ${OPENLIBM_ARCH_FOLDER} STREQUAL "aarch64")
     set(OPENLIBM_ARCH_FOLDER "aarch64")
+elseif(${OPENLIBM_ARCH_FOLDER} STREQUAL "armv7-a")
+    set(OPENLIBM_ARCH_FOLDER "arm")
 elseif(${OPENLIBM_ARCH_FOLDER} STREQUAL "x86")
     set(OPENLIBM_ARCH_FOLDER "i387")
 elseif(${OPENLIBM_ARCH_FOLDER} STREQUAL "powerpc")
@@ -428,6 +430,11 @@ elseif(${OPENLIBM_ARCH_FOLDER} STREQUAL "aarch64")
             "${PROJECT_SRC}/aarch64/fenv.c"
     )
 
+elseif(${OPENLIBM_ARCH_FOLDER} STREQUAL "arm")
+    list(APPEND OPENLIBM_C_SOURCE
+            "${PROJECT_SRC}/${OPENLIBM_ARCH_FOLDER}/fenv.c"
+    )
+
 elseif(${OPENLIBM_ARCH_FOLDER} STREQUAL "i387")
     list(APPEND OPENLIBM_C_SOURCE
             "${PROJECT_SRC}/i387/fenv.c"
